modules/sdk/project-templates/project-templates/src/test/java/com/liferay/project/templates/CodeTemplatesTest.java

```diff
@@ -0,0 +1,703 @@
+		codeTemplatesArgs.setDestinationDir(projectDir);
+		codeTemplatesArgs.setPackageName("com.liferay");
+		codeTemplatesArgs.setTemplate("rendercommand");
+
+		new CodeTemplates(codeTemplatesArgs);
+
+		File javaDir = new File(projectDir, "src/main/java");
+
+		File renderCommandFile = new File(
+			javaDir, "com/liferay/rendercommand/FooMVCRenderCommand.java");
+
+		Assert.assertTrue(renderCommandFile.exists());
+
+		_testContainsOrNot(
+			javaDir, "com/liferay/rendercommand/FooMVCRenderCommand.java",
+			false, true, "mvc.command.name=/rendercommand/foo");
+
+		if (Validator.isNotNull(ProjectTemplatesTest.BUILD_PROJECTS) &&
+			ProjectTemplatesTest.BUILD_PROJECTS.equals("true")) {
+
+			ProjectTemplatesTest.executeGradle(
+				projectDir, _GRADLE_TASK_PATH_BUILD);
+		}
+	}
+
+	@Test
+	public void testGenerateCodeTemplateResourceCommand() throws Exception {
+		File destinationDir = temporaryFolder.newFolder("code");
+
+		ProjectTemplatesArgs projectTemplatesArgs = new ProjectTemplatesArgs();
+
+		Path jarPath = FileUtil.getJarPath();
+
+		File archetypesDir = jarPath.toFile();
+
+		projectTemplatesArgs.setArchetypesDirs(
+			Collections.singletonList(archetypesDir));
+
+		projectTemplatesArgs.setDestinationDir(destinationDir);
+		projectTemplatesArgs.setLiferayVersion("7.1");
+		projectTemplatesArgs.setName("foo");
+		projectTemplatesArgs.setMaven(true);
+		projectTemplatesArgs.setTemplate("mvc-portlet");
+
+		new ProjectTemplates(projectTemplatesArgs);
+
+		File projectDir = new File(destinationDir, "foo");
+
+		CodeTemplatesArgs codeTemplatesArgs = new CodeTemplatesArgs();
+
+		codeTemplatesArgs.setAddtionalParameters(
+			Collections.unmodifiableMap(
+				Stream.of(
+					new AbstractMap.SimpleEntry<>(
+						"portletName", "ResourceCommand")
+				).collect(
+					Collectors.toMap(
+						AbstractMap.SimpleEntry::getKey,
+						AbstractMap.SimpleEntry::getValue)
+				)));
+		codeTemplatesArgs.setArchetypesDirs(
+			Collections.singletonList(archetypesDir));
+		codeTemplatesArgs.setAuthor("fred");
+		codeTemplatesArgs.setClassName("Foo");
+		codeTemplatesArgs.setDestinationDir(projectDir);
+		codeTemplatesArgs.setPackageName("com.liferay");
+		codeTemplatesArgs.setTemplate("resourcecommand");
+
+		new CodeTemplates(codeTemplatesArgs);
+
+		File javaDir = new File(projectDir, "src/main/java");
+
+		File mvcResourceCommandFile = new File(
+			javaDir, "com/liferay/resourcecommand/FooMVCResourceCommand.java");
+
+		Assert.assertTrue(mvcResourceCommandFile.exists());
+
+		_testContainsOrNot(
+			javaDir, "com/liferay/resourcecommand/FooMVCResourceCommand.java",
+			false, true, "mvc.command.name=/resourcecommand/foo");
+
+		if (Validator.isNotNull(ProjectTemplatesTest.BUILD_PROJECTS) &&
+			ProjectTemplatesTest.BUILD_PROJECTS.equals("true")) {
+
+			File buildGradle = new File(projectDir, "build.gradle");
+
+			String content = new String(
+				Files.readAllBytes(buildGradle.toPath()));
+
+			String deps =
```

Comment:
Please use full variable names instead of abbreviations for readability (i.e. `dependencies`)..